### PR TITLE
Refactored getPenpendicularFillGrowth() for simplicity

### PR DESCRIPTION
**Pull request by jaykay99992**
Refactored getPenpendicularFillGrowth() through adding a reusable function growthCalculation()

### CODE REVIEW

File: jhotdraw-core/src/main/java/org/jhotdraw/draw/AttributeKeys.java [hunk 1]
```diff
@@ -524,53 +524,41 @@ public static int getFontStyle(Figure f) {
...
-            break;
-          case OUTSIDE:
-            grow = 0f;
-            break;
-          case CENTER:
-          default:
-            grow = strokeWidth / -2d;
-            break;
-        }
+        grow = growthCalculation(placement, strokeWidth);
```
Comment (JFLeblay) [inline jhotdraw-core/src/main/java/org/jhotdraw/draw/AttributeKeys.java]: In this situation, growthCalculation will NOT return what you expect

File: jhotdraw-core/src/main/java/org/jhotdraw/draw/AttributeKeys.java [hunk 2]
```diff
@@ -524,53 +524,41 @@ public static int getFontStyle(Figure f) {
...
-            break;
-          case OUTSIDE:
-            grow = strokeWidth / 2d;
-            break;
-          case CENTER:
-          default:
-            grow = 0d;
-            break;
-        }
+        grow = growthCalculation(placement, strokeWidth);
```
Comment (JFLeblay) [inline jhotdraw-core/src/main/java/org/jhotdraw/draw/AttributeKeys.java]: Same comment as before

**Review by JFLeblay — CHANGES_REQUESTED** [review]
Whereas it is a good idea to try to refactor duplicate switch statements, the changes you have implemented are indeed changing the behavior of your method getPerpendicularFillGrowth. Now, after refactoring, the method getPerpendicularFillGrowth will always return the same value of grow regardless of the value in STROKE_PLACEMENT.
I cannot approve this Pull Request as it is. 
If you want to make the method getPerpandicularFillGrowth shorter, you could extract each `switch (placement)` into a separate method.

**Comment by jaykay99992** [comment]
@JFLeblay I just refactored the code even more through treating each case of FILL_UNDER_STROKE in a separate method, is it fine like this ? do I merge it , or just reject the whole pull request if you think it became much more complicated ?

**Review by JFLeblay — APPROVED** [review]
Now the behovior is not altered and your code is cleaner.
You can merge it.

**Comment by jaykay9999** [comment]
Sure, code merged

**Pull request merged**